content/blog/living-off-the-ipad-as-an-engineer/index.md

```diff
@@ -0,0 +1,110 @@
+
+Now even cheaper options are possible, such as just getting a good regular Bluetooth keyboard to complete a stand mount like the mechanical [Keychron K2](https://www.keychron.com/products/keychron-k2-wireless-mechanical-keyboard), as well as a mouse—I can’t *not *recommend the [MX Master series](https://www.logitech.com/en-ca/products/mice/mx-master-3.910-005620.html), but the Pebble on the pocketable side is excellent and very affordable in the [K380 combo](https://www.logitech.com/en-ca/products/combos/k380-m350-keyboard-mouse-combo.html?crid=1759). The external [Magic Trackpad](https://www.apple.com/shop/product/MRMF2LL/A/magic-trackpad-2-space-gray?fnode=77ac33807229049f26bebb326a1b4522488e4f051b1a666c9afcaabe58587a06b83e6823fcf6b2252221dfa0a67c6d6503d4048f5a20ae59be372257a04c999d1c4039a886b6c06bad71362915abc57426612fc100c35fb1e968420f659be9e0) 2 is incredible to use with iPadOS but is, granted, on the expensive edge of pointing devices.
+
+![A quite simple desk setup. Doing some LumaFusion edits!](setup.jpeg)
+
+*Update: After quite some time with Magic Keyboard for iPad on the desk, I revised the setup, with the tablet laying flat under the monitor to quickly take handwritten notes or drawings, with the Keychron K2 + Magic Trackpad 2 as input devices. We’ll see if it sticks!*
+
+## Software Dev: Remote Server and Native Apps
+
+Let’s get into real engineering tools. As you may know very well already, there’s no walled garden like iOS/iPadOS. Apps are fully contained, *Files* is some kind of explorer yet remains very limited, and firing a local command line is pure fantasy.
+
+*How the heck do we write and run our beloved code, Pierre? 💁*
+
+As it’s common these days, the trick is in the cloud. While it comes with its drawbacks such as with spotty connections, offloading the work to a remote, safe, always-accessible machine has some nice things going for it. A hosted instance can be fired in minutes these days. To stay somewhat minimalist, I chose to keep an old laptop plugged in a closet, but there are many options out there.
+
+There’s essentially two solutions that work for me at the moment: [Visual Studio Code](https://code.visualstudio.com/) in the browser, which can be achieved in various ways; or a native SSH client app such as [Termius](https://apps.apple.com/us/app/termius-ssh-client/id549039908) or [Blink Shell](https://blink.sh/). The former is a cross-platform cloud SSH/Mosh service that has really good iPad support, its main downside being the lack of a full external monitor integration, as mentioned earlier. Blink Shell features it, and it works quite well. Picture that clean tmux/vim session on a 4K monitor, at the cost of losing mouse support, though.
```

Comment:
There’s essentially two solutions that work for me at the moment: [Visual Studio Code](https://code.visualstudio.com/) in the browser, which can be achieved in various ways; or a native SSH client app such as [Termius](https://apps.apple.com/us/app/termius-ssh-client/id549039908) or [Blink Shell](https://blink.sh/). The former is a cross-platform cloud SSH/Mosh service that has really good iPad support. Its main downside is the lack of a full external monitor integration, as mentioned earlier. Blink Shell features it, and it works quite well. Picture that clean tmux/vim session on a 4K monitor, at the cost of losing mouse support, though.